closed/custom/Docs.gmk

```diff
@@ -112,7 +110,199 @@ $(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
+# 4. Find and replace the summary text also in the top level index.html file.
+
+OPENJ9_ONLY_JAVADOC_SCRIPT_BODY = \
+    for file in `$(FIND) $(OPENJ9_ONLY_DOCDIR) -type f` ; do \
+      $(CP) -f "$$file" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      $(call sed_replace_all,class="externalLink",class="externalLink" target="_blank") <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$file" ; \
+    done ; \
+    for file in `cd $(OPENJ9_ONLY_DOCDIR) ; $(FIND) . -type f -name module-summary.html | $(SED) -e 's|^\./||'` ; do \
+      fullpath=$(OPENJ9_ONLY_DOCDIR)/$$file ; \
+      dir="$${file%/*}" ; \
+      topdir="$${file%%/*}" ; \
+      excepttopdir="$${file\#*/}" ; \
+      excepttopdir=`$(SED) 's|-summary.html$$|-info.java|' <<<$$excepttopdir` ; \
+      regex="$$topdir/.*$$excepttopdir" ; \
+      if ! $(GREP) -qe "$$regex" $(OPENJ9_ONLY_JAVADOC_FILES) ; then \
+        $(CP) -f "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_block,$(MODULE_DESCRIPTION_SECOND_LINE),$(MODULE_DESCRIPTION_LAST_LINE),$(MODULE_DESCRIPTION_MARKER_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath" ; \
+        $(CP) -f "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_regex,$(MODULE_DESCRIPTION_FIRST_LINE).*$(MODULE_DESCRIPTION_MARKER_TEXT),$(MODULE_DESCRIPTION_REPLACE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath" ; \
+        $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      fi ; \
+    done ; \
+    if ! $(GREP) -q '$(OPENJDK_JDK_MANAGEMENT_SUMMARY)' $(OPENJ9_ONLY_INDEX_HTML) ; then \
+      $(ECHO) "$$0: $(OPENJDK_JDK_MANAGEMENT_SUMMARY_ERROR_TEXT)" ; \
+      exit 1 ; \
+    else \
+      $(CP) -f "$(OPENJ9_ONLY_INDEX_HTML)" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      $(call sed_replace_first,$(OPENJDK_JDK_MANAGEMENT_SUMMARY),$(OPENJDK_MODULE_REFERENCE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$(OPENJ9_ONLY_INDEX_HTML)" ; \
+      $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+    fi ; \
+    #
+
+OPENJ9_ONLY_JAVADOC_SCRIPT_BODY_ESC = $(subst ','\'',$(OPENJ9_ONLY_JAVADOC_SCRIPT_BODY))
+
+# Create a stub package-info.java file for each package with source files in
+# j9jcl_sources but no package-info.java file. This is then used by javadoc
+# instead of the openjdk version to direct the user to the Oracle online
+# documentation for that package.
+
+# The dependency on $(OPENJDK_JAVADOC_MARKER) ensures that the docs/api/element-list
+# file which is created during the full jdk javadoc generation exists before running
+# javadoc for just the openj9 files.
+
+# If running on Windows, The source file names in the response file must be converted to
+# Windows format before passing them to javadoc.
+OPENJ9_ONLY_JAVADOC_FIXPATH_FILELIST := $(foreach file,$(OPENJ9_ONLY_JAVADOC_FILELIST),$(call FixPath,$(file)))
+
+$(OPENJ9_ONLY_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER) $(OPENJ9_ONLY_FILELIST)
+	$(call LogWarn, Generating javadoc for openj9 source files)
+	$(ECHO) '$(OPENJ9_ONLY_JAVADOC_SCRIPT_BODY_ESC)' >$(OPENJ9_ONLY_JAVADOC_SCRIPT)
```

Comment:
In the previous version, this line was immediately before the use by bash (now line 306). I don't think it's wrong here - just curious why it moved?

Reply:
Ah yes.
I moved it there so that the file OPENJ9_ONLY_JAVADOC_FILES would be written to the script file with its cygwin file name because I was then converting that file name to Windows format before passing it to javadoc on the javadoc command line. After adding some logging and examining the output I realised that windows command lines are actually run via fixpath.exe which (I now assume) converts all cygwin paths in the command line to windows format before executing the command.  So the the file name conversion I had done was unnecessary - and I then forgot to put the line writing out the script code back to where it was originally.

Reply:
Thanks for the explanation. I'll let you decide whether you want to move it back or not.

Reply:
I prefer it in its original place, immediately before the line which executes the script.  I've updated the PR.